src/arrayMethodSplice.js

```diff
@@ -5,7 +5,188 @@
+    if (howManyDelete <= 0 && addItems.length !== 0) {
+      for (let i = this.length - 1; i >= startIndex; i--) {
+        this[i + addItems.length] = this[i];
+      }
+
+      for (let i = 0; i < addItems.length; i++) {
+        this[i + startIndex] = addItems[i];
+      }
+
+      return [];
+    }
+
+    // case if only start point received
+    if (howManyDelete === undefined && addItems.length === 0) {
+      let v = 0;
+
+      for (let i = startIndex; i < this.length; i++) {
+        removedValues[v] = this[i];
+        v++;
+      }
+      this.length -= removedValues.length;
+
+      return removedValues;
+    }
+
+    // case if nothing needs to add:
+    // 1) taking deleted values
+    if (addItems.length === 0) {
+      for (let i = 0; i < howManyDelete; i++) {
+        removedValues[i] = this[startIndex + i];
+      }
+
+      const afterArr = [];
+      let v = 0;
+
+      // 2) saving values after deleted ones
+      for (let i = startIndex + howManyDelete; i < this.length; i++) {
+        afterArr[v] = this[i];
+        v++;
+      }
+
+      // 3) putting 'after' values instead deleted ones
+      for (let i = startIndex; i < this.length; i++) {
+        this[i] = afterArr[i - startIndex];
+      }
+
+      this.length -= removedValues.length;
+
+      return removedValues;
+    }
+
+    // case if we need to add more elements than delete:
+    // 1) taking deleted values
+    if (addItems.length > howManyDelete) {
+      for (let i = 0; i < howManyDelete; i++) {
+        removedValues[i] = this[startIndex + i];
+      }
+
+      const afterArr = [];
+      let v = 0;
+
+      // 2) saving values after deleted ones
+      for (let i = startIndex + howManyDelete; i < this.length; i++) {
+        afterArr[v] = this[i];
+        v++;
+      }
+
+      // 3) adding extra length for more elements
+      this.length += addItems.length - howManyDelete;
+
+      // 4) adding new values
+      for (let i = startIndex; i < startIndex + addItems.length; i++) {
+        this[i] = addItems[i - startIndex];
+      }
+
+      // 5) putting back 'after' values
+      for (let i = startIndex + addItems.length; i < this.length; i++) {
+        this[i] = afterArr[i - (startIndex + addItems.length)];
+      }
+
+      return removedValues;
+    }
+
+    // case if we need to remove as much as add:
+    if (addItems.length === howManyDelete) {
+      for (let i = 0; i < howManyDelete; i++) {
+        removedValues[i] = this[startIndex + i];
+      }
+
+      const afterArr = [];
+      let v = 0;
+
+      for (let i = startIndex + howManyDelete; i < this.length; i++) {
+        afterArr[v] = this[i];
+        v++;
+      }
+
+      for (let i = startIndex; i < startIndex + addItems.length; i++) {
+        this[i] = addItems[i - startIndex];
+      }
+
+      for (let i = startIndex + addItems.length; i < this.length; i++) {
+        this[i] = afterArr[i - (startIndex + addItems.length)];
+      }
```

Comment:
Seems that I've seen this logic before, let's find regularity and move to separate function 

You can pass this as parameter
You can reuse same loop also if there are no items to add (`startIndex + 0 !< startIndex` - loop won't start)